### PR TITLE
Translation function and t-translation attribute

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: src/qweb/compiler.ts [hunk 1]
```diff
@@ -28,8 +29,8 @@ export type TemplateFunction = (blocks: any, utils: any) => Template;
 
 type BlockType = "block" | "text" | "multi" | "list" | "html";
 
-export function compileTemplate(template: string, name?: string): TemplateFunction {
-  const compiler = new QWebCompiler(template, name);
+export function compileTemplate(template: string, name?: string, translateFn?: (s: string) => string): TemplateFunction {
```
Comment (Polymorphe57) [inline src/qweb/compiler.ts:32]: don't like that API

Comment (ged-odoo) [reply src/qweb/compiler.ts:32]: probably better to put the name and the translateFn as option

File: src/qweb/compiler.ts [hunk 2]
```diff
@@ -143,11 +152,13 @@ export class QWebCompiler {
...
   templateName: string;
   template: string;
+  translateFn?: (s: string) => string;
   ast: AST;
   staticCalls: { id: string; template: string }[] = [];
 
-  constructor(template: string, name?: string) {
+  constructor(template: string, name?: string, translateFn?: (s: string) => string) {
     this.template = template;
+    this.translateFn = translateFn;
```
Comment (ged-odoo) [inline src/qweb/compiler.ts:161]: would probably be better to have a fallback: `this.translateFn = translateFn || word => word;` so you don't need to check later on if it exists

**Comment by Laris15** [comment]
Hello everyone, please i have an issue in odoo 14.
I have translated a module i created from English to French by exporting the fr.po file and translating each term in it. But i dont understand why when i change my Odoo language to French all my module is still in English. 

Please can anyone help me

**Pull request merged**